crates/miden-lib/asm/miden/note.masm

```diff
@@ -45,75 +54,80 @@ export.get_assets
     # => [num_assets, dest_ptr]
 end
 
-#! Loads the note's inputs to `dest_ptr`.
+#! Returns the recipient of the note currently being processed.
+#!
+#! Inputs:  []
+#! Outputs: [RECIPIENT]
+#!
+#! Where:
+#! - RECIPIENT is the commitment to the input note's script, inputs, the serial number.
+#!
+#! Panics if:
+#! - no note is being processed.
+#!
+#! Invocation: exec
+export.get_recipient
+    # pad the stack
+    padw padw padw push.0.0
+    # => [pad(14)]
+
+    # push the flag indicating that we want to request recipient from the currently processing
+    # note
+    push.1
+    # => [is_current_note = 1, pad(14)]
+
+    exec.kernel_proc_offsets::input_note_get_recipient_offset
+    # => [offset, is_current_note = 1, pad(14)]
+
+    syscall.exec_kernel_proc
+    # => [RECIPIENT, pad(12)]
+
+    # clean the stack
+    swapdw dropw dropw swapw dropw
+    # => [RECIPIENT]
+end
+
+#! Writes the note's inputs to `dest_ptr`.
```

Comment:
I would write this as "Writes the active note's inputs to memory starting at `dest_ptr`."